dvc/repo/experiments/pull.py

```diff
@@ -85,7 +85,7 @@ def _pull(
 
     with TqdmGit(desc="Fetching git refs") as pbar:
         results: Mapping[str, SyncStatus] = repo.scm.fetch_refspecs(
-            git_remote,
```

Comment:
This should really be handled in scmrepo, and `scmrepo.scm.fetch_refspecs` should accept both URLs and git remote names. The issue is that the SSH check should be handled in `pygit2._fetch_refspecs` once we actually load a remote (and it should raise `NotImplementedError` for URLs the backend can't handle.

So probably here:
https://github.com/iterative/scmrepo/blob/b5c804e5967209dc5ec2fa8a8fbb0ba5c0098ab7/src/scmrepo/git/backend/pygit2.py#L463-L466

Reply:
I did it this way because of https://github.com/iterative/scmrepo/blob/b5c804e5967209dc5ec2fa8a8fbb0ba5c0098ab7/src/scmrepo/git/backend/base.py#L231-L245

Reply:
The docstring is wrong, it should have been updated at the same time as `push_refspec` in https://github.com/iterative/scmrepo/commit/0d0d92e0d1919d1bcc87a256a567c520e97eaa4f